src/main/resources/web/app/qs-form.ts

```diff
@@ -0,0 +1,174 @@
+export const QS_RESULT_EVENT = 'qs-result';
+export const QS_RESULT_NEXT_EVENT = 'qs-result-next';
+
+export const QS_NEXT_PAGE_EVENT = 'qs-next-page';
+
+export interface QsResult {
+  hits: QsHit[];
+  hasMoreHits: boolean;
+}
+
+export interface QsHit {
+  title: string;
+  summary: string;
+  url: string;
+  keywords: string | undefined;
+  content: string | undefined;
+  type: string | undefined;
+}
+
+
+@customElement('qs-form')
+export class QsForm extends LitElement {
+
+  static styles = css`
+    .quarkus-search {
+      display: block !important;
+    }
+
+    .d-none {
+      display: none;
+    }
+  `;
+
+  @property({type: String}) api: string = "/api/search";
+  @property({type: String}) minChars: number = 3;
+  @property({type: String}) quarkusVersion: string;
+
+  private _page: number = 0;
+  private _total: number = 0;
+  private _loading: boolean = false;
+
+  render() {
+    return html`
+      <div id="qs-form">
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    const formElements = this._getFormElements();
+    this.addEventListener(QS_NEXT_PAGE_EVENT, this._handleNextPage);
+    formElements.forEach((el) => {
+      const eventName = this._isInput(el) ? 'input' : 'change';
+      el.addEventListener(eventName, this._handleInputChange);
+    });
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener(QS_NEXT_PAGE_EVENT, this._handleNextPage);
+    const formElements = this._getFormElements();
+    formElements.forEach(el => {
+      const eventName = this._isInput(el) ? 'input' : 'change';
+      el.removeEventListener(eventName, this._handleInputChange);
+    });
+  }
+
+  private _getFormElements(): NodeListOf<HTMLFormElement> {
+    return this.querySelectorAll('input, select');
+  }
+
+  private _readQueryInputs() {
+    const formElements = this._getFormElements();
+    const formData = {};
+
+    formElements.forEach((el: HTMLFormElement) => {
+      if (el.value && el.value.length > 0) {
+        formData[el.name] = el.value;
+      }
+    });
+
+    console.log(JSON.stringify(formData));
+    return formData;
+  }
+
+  private _search = () => {
+    if (this._loading) {
+      return;
+    }
+    this.dispatchEvent(new CustomEvent(QS_START_EVENT));
+    const data = this._readQueryInputs();
+    this._loading = true;
+    this._jsonFetch(new AbortController(), 'GET', data, 1000)
+      .then((r: any) => {
+        let event = QS_RESULT_EVENT
+        if (this._page > 0 && r.hits.length > 0) {
+          this._total += r.hits.length;
+          event = QS_RESULT_NEXT_EVENT;
+        } else {
+          this._total = r.hits.length;
+        }
+        const hasMoreHits = r.total > this._total;
+
+        this.dispatchEvent(new CustomEvent(event, {detail: {...r, hasMoreHits}}));
+      }).finally(() => {
+      this._loading = false;
+      this.dispatchEvent(new CustomEvent(QS_END_EVENT));
+    });
```

Comment:
I don't see offline (javascript) search handling in case we reach the timeout... Is it supposed to be implemented elsewhere?

Reply:
Yes, I need to see if we pass the data with json or if we need to parse the page content and send an event for this.